src/volue/mesh/aio/_connection.py

```diff
@@ -379,6 +381,64 @@ async def update_rating_curve_versions(
             )
             await self.mesh_service.UpdateRatingCurveVersions(request)
 
+        async def run_simulation(
+            self,
+            model: str,
+            case: str,
+            start_time: datetime,
+            end_time: datetime,
+            resolution: Timeseries.Resolution,
+            scenario: int,
+            return_datasets: bool,
+        ) -> typing.AsyncIterator[None]:
+            if start_time is None or end_time is None:
+                raise TypeError("start_time and end_time must both have a value")
+
+            case_group, case_name = case.split("/", maxsplit=1)
+
+            simulation = f"Model/{model}/{case_group}.has_OptimisationCases/{case_name}.has_OptimisationParameters/Optimal.has_HydroSimulation/HydroSimulation"
+
+            request = core_pb2.SimulationRequest(
+                session_id=_to_proto_guid(self.session_id),
+                simulation=_to_proto_object_mesh_id(simulation),
+                interval=_to_proto_utcinterval(start_time, end_time),
+                scenario=scenario,
+                return_datasets=return_datasets,
+            )
```

Comment:
You can move the code for checking timestamps, splitting case group and name and request preparation to some `_prepare_simulation_request`. We are doing something like this for some other RPCs. I'm all for not repeating the code if possible, even if it means to go one level deeper to find out the logic. The same for inflow calculation. Not a blocker though.

Reply:
Will resolve in a follow-up PR, I'm planning to do the `_base_session.py` thing as well.